spec/ttt/game_rules_spec.clj

```diff
@@ -0,0 +1,53 @@
+(ns ttt.game-rules-spec
+  (:require [speclj.core :refer [describe context it should=]]
+            [ttt.game-rules :as rules]))
+
+
+(describe "Game Rules"
+  (context "#win?"
+    (it "returns false if there is no winner"
+      (should= false (rules/win? ["X" "X"  "O"
+                                  "O" "O"  "X"
+                                   6   7   "X"])))
+    (it "checks if the game is won for a row of X's"
+      (should= true (rules/win? ["X" "X" "X"
+                                 "O"  4  "O"
+                                  5   6   7])))
+    (it "checks if the game is won for a row of O's"
+      (should= true (rules/win? ["X"  1  "X"
+                                 "O" "O" "O"
+                                  5   6   7])))
+    (it "checks if the game is won for a column of X's"
+      (should= true (rules/win? ["X" "O" "O"
+                                 "X"  4  "O"
+                                 "X"  6   7])))
+    (it "checks if the game is won for a column of O's"
+      (should= true (rules/win? ["X" "O" "X"
+                                 "X" "O" "O"
+                                  5  "O"  7])))
+    (it "checks if the game is won for a diagonal of X's"
+      (should= true (rules/win? ["O" "X" "X"
+                                 "O" "X" "O"
+                                 "X"  6   7])))
+    (it "checks if the game is won for a diagonal of O's"
+      (should= true (rules/win? ["O" "X" "X"
+                                  3  "O" "O"
+                                 "X"  6  "O"]))))
+  (context "#tie?"
```

Comment:
It looks like you may have missed a case for `rules/tie?`: when there aren't any available spaces, and there is a winner.  If I'm not mistaken, that means this incorrect version of `rules/tie?` might still pass the tests here

```clojure
(defn rules/tie?
  [board]
  (not (available-spaces? board)))
```

This in turn makes me wonder: were these tests written after you already implemented `rules/tie?`?

Reply:
That does happen from time to time: you thought you were taking the smallest possible leap in the production code that got your tests to pass, only to realize later you could have taken a smaller step and added an additional test.  So it's all good; seeing gaps is a good way to make a reasonable guess about the process that was used to create the code.

You can implement `win?` and `tie?` by having one use the other for (part of) its algorithm.  Either way is fine in my view; just make sure you don't have them pointing to each other (you'll get an endless loop).